src/Orchard.Web/Modules/Orchard.Tokens/Providers/ListTokens.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Web;
+using Orchard.ContentManagement;
+using Orchard.Localization;
+using System.Linq;
+	 
+namespace Orchard.Tokens.Providers {
+	 
+	public class ListTokens : ITokenProvider {
+	    private readonly Func<ITokenizer> _tokenizer;
+	 
+	    public ListTokens(Func<ITokenizer> tokenizer) {
+	        _tokenizer = tokenizer;
+	        T = NullLocalizer.Instance;
+	    }
+	 
+	    public Localizer T { get; set; }
+	 
+	    public void Describe(DescribeContext context) {
+	        context.For("List", T("Lists"), T("Handles lists of Content Items"))
+                .Token("Join:*", T("Join:<Tokenized text>[,<separator>]"), T("Join each element in the list by applying the tokenized text and concatenating the output with the optional separator."))
```

Comment:
I don't really like the syntax, will think about it, or let others suggest something better.